### PR TITLE
Debugger: Add string comparison to conditional breakpoints

### PR DESCRIPTION
**Pull request by SeekyCt**
Adds a basic `streq` function to conditional breakpoints for comparison of strings in RAM and/or string constants. It takes two parameters, either of which can be an address or a string constant.

Examples of valid calls:
- `streq(r3, "abc")`
- `streq(0x8000000, r3)`
- `streq("abc", 0x80000000)`
- `streq("abc", "abc")`

### CODE REVIEW

**Comment by sepalani** [comment]
@dolphin-emu-bot rebuild

**Comment by sepalani** [comment]
@dolphin-emu-bot rebuild

File: Source/Core/Core/PowerPC/Expression.cpp [hunk 1]
```diff
@@ -131,7 +131,43 @@ static double CallstackFunc(expr_func* f, vec_expr_t* args, void* c)
...
+    return 0;
+
+  std::array<std::string, 2> strs;
+  for (int i = 0; i < 2; i++)
+  {
+    std::optional<std::string> arg = ReadStringArg(&vec_nth(args, i));
+    if (arg == std::nullopt)
+      return 0;
+
+    strs[i] = arg.value();
```
Comment (lioncash) [inline Source/Core/Core/PowerPC/Expression.cpp:164]:
```suggestion
    strs[i] = std::move(*arg);
```

**Comment by AdmiralCurtiss** [comment]
@dolphin-emu-bot rebuild

**Comment by AdmiralCurtiss** [comment]
Rebased this to fix a conflict that snuck in due to #11554. Seems to work. I'm getting some spurious pauses with conditional breakpoints but that doesn't seem to be this PRs fault.

**Pull request merged**